Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -98,30 +154,23 @@ async function doWork() {
         }
         tl.debug(tl.loc('GetArtifactsFromBuildNumber', buildId, jobName));
 
-        const artifactQueryUrl: string = `${serverEndpointUrl}/job/${jobName}/${buildId}/api/json?tree=artifacts[*]`;
-        console.log(tl.loc('ArtifactDownloadUrl', artifactQueryUrl));
-
-        var templatePath = path.join(__dirname, 'jenkins.handlebars.txt');
-        var variables = {
-            "endpoint": {
-                "url": serverEndpointUrl
-            },
-            "definition": jobName,
-            "version": buildId
-        };
-        var handler = new handlers.BasicCredentialHandler(username, password);
-        
-        var webProvider = new providers.WebProvider(artifactQueryUrl, templatePath, variables, handler, { ignoreSslError: !strictSSL });
-        var localFileProvider = new providers.FilesystemProvider(localPathRoot);
-    
-        let downloader = new engine.ArtifactEngine();
-        var downloaderOptions = new engine.ArtifactEngineOptions();
-        downloaderOptions.itemPattern = tl.getInput('itemPattern', false) || "**";
-        downloaderOptions.parallelProcessingLimit = +tl.getVariable("release.artifact.download.parallellimit") || 4;
-        var debugMode = tl.getVariable('System.Debug');
-        downloaderOptions.verbose = debugMode ? debugMode.toLowerCase() != 'false' : false;
-
-        await downloader.processItems(webProvider, localFileProvider, downloaderOptions);
+        if (tl.getBoolInput("propagatedArtifacts") == true) {
+            var containerName: string = tl.getInput("containerName");
+            var commonVirtualPath: string = tl.getInput("commonVirtualPath");
+            getArtifactsFromAzureRMStorage(localPathRoot, containerName, commonVirtualPath);
+        } else {
+            const artifactQueryUrl: string = `${serverEndpointUrl}/job/${jobName}/${buildId}/api/json?tree=artifacts[*]`;
+            var variables = {
+                "endpoint": {
+                    "url": serverEndpointUrl
+                },
+                "definition": jobName,
+                "version": buildId
+            };
+            var handler = new handlers.BasicCredentialHandler(username, password);
+
+            getArtifactsFromUrl(artifactQueryUrl, strictSSL, localPathRoot, itemPattern, handler, variables);
```

Comment:
consider having the logic for azure storage in a separate file. This will make the code here cleaner